ports/mimxrt10xx/peripherals/mimxrt10xx/MIMXRT1062/clocks.c

```diff
@@ -335,3 +336,270 @@ void clocks_init(void) {
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * 1. The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * 2. If the Software is incorporated into a build system that allows
+ * selection among a list of target devices, then similar target
+ * devices manufactured by PJRC.COM must be included in the list of
+ * target devices and selectable in the same manner.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+// uint32_t set_arm_clock(uint32_t frequency);
+
+// stuff needing wait handshake:
+//  CCM_CACRR  ARM_PODF
+//  CCM_CBCDR  PERIPH_CLK_SEL
+//  CCM_CBCMR  PERIPH2_CLK_SEL
+//  CCM_CBCDR  AHB_PODF
+//  CCM_CBCDR  SEMC_PODF
+
+uint32_t setarmclock(uint32_t frequency) {
+    uint32_t cbcdr = CCM->CBCDR;     // pg 1021
+    uint32_t cbcmr = CCM->CBCMR;     // pg 1023
+    uint32_t dcdc = DCDC->REG3;
+
+    // compute required voltage
+    uint32_t voltage = 1150;     // default = 1.15V
+    if (frequency > 528000000) {
+        voltage = 1250;         // 1.25V
+        #if defined(OVERCLOCK_STEPSIZE) && defined(OVERCLOCK_MAX_VOLT)
+        if (frequency > 600000000) {
+            voltage += ((frequency - 600000000) / OVERCLOCK_STEPSIZE) * 25;
+            if (voltage > OVERCLOCK_MAX_VOLT) {
+                voltage = OVERCLOCK_MAX_VOLT;
+            }
+        }
+        #endif
+    } else if (frequency <= 24000000) {
+        voltage = 950;         // 0.95
+    }
+
+    // if voltage needs to increase, do it before switch clock speed
+    CCM->CCGR6 |= CCM_CCGR6_DCDC(CCM_CCGR_ON);
+    if ((dcdc & ((uint32_t)(0x1F << 0))) < ((uint32_t)(((voltage - 800) / 25) & 0x1F) << 0)) {
+        // printf("Increasing voltage to %u mV\n", voltage);
+        dcdc &= ~((uint32_t)(0x1F << 0));
+        dcdc |= ((uint32_t)(((voltage - 800) / 25) & 0x1F) << 0);
+        DCDC->REG3 = dcdc;
+        while (!(DCDC->REG0 & DCDC_REG0_STS_DC_OK_L)) {
+            ;                                                   // wait voltage settling
+        }
+    }
+
+    if (!(cbcdr & CCM_CBCDR_PERIPH_CLK_SEL_L)) {
+        // printf("need to switch to alternate clock during reconfigure of ARM PLL\n");
```

Comment:
Those are primarily debug comments.  I left them commented in case I wanted to go back and do some debugging if things didn't work.  Same for all the printf in clocks.c

Reply:
Just fixed all the printf.